session/session.go

```diff
@@ -3,16 +3,35 @@ package session
 import (
 	"crypto/aes"
 	"crypto/cipher"
+	"crypto/hmac"
 	"crypto/sha1"
 	"encoding/base64"
+	"encoding/hex"
+	"errors"
 	"net/url"
 	"strings"
 
 	"golang.org/x/crypto/pbkdf2"
 )
 
 func generateSecret(base, salt string) []byte {
-	return pbkdf2.Key([]byte(base), []byte(salt), key_iter_num, key_size, sha1.New)
+	return pbkdf2.Key([]byte(base), []byte(salt), keyIterNum, keySize, sha1.New)
+}
+
+// I stolen the snippet from: https://gist.github.com/doitian/2a89dc9e4372e55335c9111f576b47bf
```

Comment:
Probably it would be better to rephrase it to be something like this:
```go
// The origin of this snippet can be found at https://gist.github.com/doitian/2a89dc9e4372e55335c9111f576b47bf
```
😉